### PR TITLE
Add error message for set accessor visibility issues

### PR DESCRIPTION
**Pull request by weswigham**
This fixes the crash in `RWC`.

### CODE REVIEW

File: tests/baselines/reference/accessorDeclarationEmitVisibilityErrors.errors.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+tests/cases/compiler/accessorDeclarationEmitVisibilityErrors.ts(2,18): error TS2304: Cannot find name 'DoesNotExist'.
+tests/cases/compiler/accessorDeclarationEmitVisibilityErrors.ts(2,18): error TS4106: Parameter 'arg' of accessor has or is using private name 'DoesNotExist'.
+
+
+==== tests/cases/compiler/accessorDeclarationEmitVisibilityErrors.ts (2 errors) ====
+    export class Q {
+        set bet(arg: DoesNotExist) {}
+                     ~~~~~~~~~~~~
```
Comment (sandersn) [inline tests/baselines/reference/accessorDeclarationEmitVisibilityErrors.errors.txt]: names that don't exist create a "is using private name" errors? ok..

Comment (weswigham) [reply tests/baselines/reference/accessorDeclarationEmitVisibilityErrors.errors.txt]: Eeeeyup.

**Comment by weswigham** [comment]
> Code looks good but it would be nice to have tests for all 3 new errors.

Yeah, unfortunately, the other two are there to cover the space of errors allowed by the type declarations for accessibility errors, but I don't think they're possible to ever trigger as, at least right now, a `set`ter cannot have inferred types. We have a lot of declaration emit errors like that, though (since we started serializing non-exported declarations if they're referenced by exported ones), so that's not so bad.

**Comment by weswigham** [comment]
And now to wait for the next rwc diff on master, which'll include a ton of `set`/`get` declarations ❤️

**Pull request merged**